packages/cli-kit/src/private/node/ui/components/ConcurrentOutput.tsx

```diff
@@ -77,48 +78,41 @@ const ConcurrentOutput: FunctionComponent<ConcurrentOutputProps> = ({
   showTimestamps = true,
   onInput,
   footer,
+  keepOpenAfterStopping,
 }) => {
   const [processOutput, setProcessOutput] = useState<Chunk[]>([])
-  const concurrentColors: TextProps['color'][] = ['yellow', 'cyan', 'magenta', 'green', 'blue']
+  const {exit: unmountInk} = useApp()
   const prefixColumnSize = Math.max(...processes.map((process) => process.prefix.length))
   const {isRawModeSupported} = useStdin()
   const [state, setState] = useState<ConcurrentOutputState>(ConcurrentOutputState.Running)
 
-  function lineColor(index: number) {
-    const colorIndex = index < concurrentColors.length ? index : index % concurrentColors.length
-    return concurrentColors[colorIndex]!
-  }
-
-  const writableStream = (process: OutputProcess, index: number) => {
-    return new Writable({
-      write(chunk, _encoding, next) {
-        const lines = stripAnsi(chunk.toString('utf8').replace(/(\n)$/, '')).split(/\n/)
-        addOrUpdateConcurrentUIEventOutput({prefix: process.prefix, index, output: lines.join('\n')}, {footer})
-
-        setProcessOutput((previousProcessOutput) => [
-          ...previousProcessOutput,
-          {
-            color: lineColor(index),
-            prefix: process.prefix,
-            lines,
-          },
-        ])
-
-        next()
-      },
-    })
-  }
-
-  const runProcesses = () => {
-    return Promise.all(
-      processes.map(async (process, index) => {
-        const stdout = writableStream(process, index)
-        const stderr = writableStream(process, index)
-
-        await process.action(stdout, stderr, abortSignal)
-      }),
-    )
-  }
+  const writableStream = useCallback(
+    (process: OutputProcess, index: number) => {
+      const concurrentColors: TextProps['color'][] = ['yellow', 'cyan', 'magenta', 'green', 'blue']
+      function lineColor(index: number) {
```

Comment:
I would not move this function and `concurrentColors` inside as they don't need to be instantiated for every writableStream. You can convert them to `useCallback` and `useMemo`.